spec/requests/client_visibility_spec.rb

```diff
@@ -5,839 +5,896 @@
+    [true, false], # project type override
+    GrdaWarehouse::Config.family_calculation_methods.values, # family calculation method
+    GrdaWarehouse::Config.available_release_durations, # release duration
+  )
 
-    describe 'and the user does not have a role' do
-      it 'user cannot see any clients' do
-        get clients_path(q: 'bob')
-        expect(response).to redirect_to(new_user_session_path)
-        # expect(response.body).to include('Sorry you are not authorized to do that.')
-        # expect(response).to have_http_status(200)
-      end
-    end
-    describe 'and the user has a role granting can view clients' do
-      before do
-        user.roles << can_view_clients
-        user.roles << can_search_window
-        sign_in user
-      end
-      it 'user can see all clients' do
-        get clients_path(q: 'bob')
-        doc = Nokogiri::HTML(response.body)
-        expect(doc.text).to include('Displaying all 2 clients')
-        expect(response).to have_http_status(200)
-      end
-    end
-    describe 'and the user has a role granting can view window clients' do
-      before do
-        user.roles << can_view_client_window
-        user.roles << can_search_window
-        sign_in user
-      end
-      it 'user can see only window clients' do
-        get clients_path(q: 'bob')
-        doc = Nokogiri::HTML(response.body)
-        expect(doc.text).to include('Displaying 1 client')
-        expect(response).to have_http_status(200)
-      end
-      it 'user can see client dashboard for window client' do
-        get client_path(window_destination_client)
-        expect(response).to have_http_status(200)
-      end
-      it 'user cannot see client dashboard for non-window client' do
-        expect do
-          get client_path(non_window_destination_client)
-          # In test environments we don't get pretty 404s, we get exceptions
-        end.to raise_error(ActiveRecord::RecordNotFound)
-      end
-    end
-    describe 'and the user has a role granting can search window' do
+  configs_variations.each do |cfg_01, cfg_02, cfg_03, cfg_04|
+    context 'when config b is in affect' do
       before do
-        user.roles << can_search_window
-        sign_in user
-      end
-      it 'user can see only window clients' do
-        get clients_path(q: 'bob')
-        doc = Nokogiri::HTML(response.body)
-        expect(doc.text).to include('Displaying 1 client')
-        expect(response).to have_http_status(200)
-      end
-      it 'user cannot see client dashboard' do
-        get client_path(window_destination_client)
-        expect(response).to redirect_to(user.my_root_path)
+        GrdaWarehouse::Config.delete_all
+        GrdaWarehouse::Config.invalidate_cache
+      end
+      # i dont know how to make rubocop happy with this
+      # rubocop:disable Layout/ArgumentAlignment
+      let!(:config) do
+        create :config_b,
```

Comment:
Note that this is using `:config_b` from `spec/factories/grda_warehouse/config.rb`.  Later on there are other `create :config_x` that should use the appropriate configs, but all are currently set to `:config_b`.